src/hex_tarball.erl

```diff
@@ -601,6 +616,12 @@ gzip_no_header(Uncompressed) ->
 %% Helpers
 %%====================================================================
 
+%% @private
+valid_size(Binary, undefined) when is_binary(Binary) ->
```

Comment:
another idea is to call the sentinel value `infinity`:

```suggestion
valid_size(Binary, infinity) when is_binary(Binary) ->
```

but perhaps undefined is more often used for things like this.

Reply:
I like `infinity` more 👍